src/main.rs

```diff
@@ -32,18 +46,57 @@ enum Command {
 
 main!(|args: Cli, log_level: verbosity| match args.cmd {
     Command::Init { path } => {
+        let started = Instant::now();
+
         let crate_path = match path {
             Some(p) => p,
             None => ".".to_string(),
         };
+
+        println!(
+            "{} {}Adding WASM target...",
+            style("[1/7]").bold().dim(),
+            TARGET
+        );
         build::rustup_add_wasm_target();
+        println!(
+            "{} {}Compiling to WASM...",
+            style("[2/7]").bold().dim(),
+            CYCLONE
+        );
         build::cargo_build_wasm(&crate_path);
+        println!(
+            "{} {}Creating a pkg directory...",
+            style("[3/7]").bold().dim(),
+            FOLDER
+        );
         wasm_pack::create_pkg_dir(&crate_path)?;
+        println!(
+            "{} {}Writing a package.json...",
+            style("[4/7]").bold().dim(),
+            MEMO
+        );
         manifest::write_package_json(&crate_path)?;
+        println!(
+            "{} {}Copying over your README...",
+            style("[5/7]").bold().dim(),
+            DANCERS
+        );
         readme::copy_from_crate(&crate_path)?;
+        println!(
+            "{} {}Installing WASM-bindgen...",
```

Comment:
Maybe we can make this an optional thing at some point? like if it's installed we just do 6 things rather than 7. But that's like a future improvement and not something that needs to be done now and block 0.1

Reply:
so currently, if wasm-bindgen is already installed this step doesn't actually do anything (it actually fails and yells in stderr). we should def check if it's installed and just skip if it is... both this and adding the target should just not happen if they are already present.

Reply:
In which case the numbering will also need to be dynamically done rather than it being hardcoded. Maybe a step at the beginning to calculate what can be done without then run the rest.